pkg/vulnerabilityreport/ttl_report_test.go

```diff
@@ -0,0 +1,124 @@
+package vulnerabilityreport_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/aquasecurity/trivy-operator/pkg/apis/aquasecurity/v1alpha1"
+	"github.com/aquasecurity/trivy-operator/pkg/ext"
+	"github.com/aquasecurity/trivy-operator/pkg/operator/etc"
+	"github.com/aquasecurity/trivy-operator/pkg/trivyoperator"
+	"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport"
+)
+
+const (
+	vreport = "vreport"
+	ns      = "default"
+)
+
+func TestRegenerateReportIfExpired(t *testing.T) {
+	// clock
+	clock := ext.NewSystemClock()
+
+	// scheme
+	scheme := trivyoperator.NewScheme()
+
+	// set the VulnerabilityScannerReportTTL
+	config, err := etc.GetOperatorConfig()
+	require.NoError(t, err)
+	hours, err := time.ParseDuration("24h")
+	require.NoError(t, err)
+	config.VulnerabilityScannerReportTTL = &hours
+
+	// logger object
+	logger := log.Log.WithName("testing")
+
+	// create TTLReport controller
+	instance := vulnerabilityreport.TTLReportReconciler{Logger: logger, Config: config, Clock: clock}
+
+	tests := []struct {
+		name                  string
+		reportUpdateTimestamp time.Duration
+		ttlAnnotationMissing  bool
+		expectError           bool
+		reportDeleted         bool
+		ttlStr                string
+		invalidReportName     bool
+	}{
+		{
+			name:                  "Report timestamp < TTL",
+			reportUpdateTimestamp: -15 * 60 * time.Minute, // < 24h TTL
+			expectError:           false,
+			reportDeleted:         false,
+			ttlStr:                "24h",
+		},
+		{
+			name:                  "Report timestamp exceeds TTL",
+			reportUpdateTimestamp: -25 * 60 * time.Minute, // > 24 TTL
+			expectError:           false,
+			reportDeleted:         true, // = time.Duration(0)
+			ttlStr:                "24h",
+		},
+		{
+			name:                 "missing TTL annotation in the report",
+			ttlAnnotationMissing: true,
+			expectError:          false, // Ignoring report without TTL set
+		},
+		{
+			name:        "invalid TTL in the annotation",
+			ttlStr:      "badtime",
+			expectError: true,
+		},
+		{
+			name:              "invalid report name",
```

Comment:
I think this is more like non-existing report. And I don't think you need an extra property for it. Can't you just use the name?

Reply:
yes fixed the name. extra property is required to set the name to an empty string.

Reply:
updated the code to explicitly set the name